src/option.bif

```diff
@@ -18,7 +18,7 @@ static bool call_option_handlers_and_set_value(StringVal* name, ID* i, Val* val,
 			bool add_loc = handler_function->FType()->AsFuncType()->ArgTypes()->Types()->length() == 3;
 			val_list vl(2 + add_loc);
 			vl.push_back(name->Ref());
-			vl.push_back(val);
+			vl.push_back(val->Ref());
```

Comment:
This change doesn't look right and likely explains at least some CI memcheck failures reporting new leaks.  The function call does consume a reference, but it also returns a new `Val` directly into `val` on return that we take ownership of -- either the next iteration consumes that or the `Unref(val)` at the end of the function will if no more iterations.

Reply:
You're right, the code is indeed correct after all! This time, my confusion was just my confusion and not a bug...